rll/collectd.sh

```diff
@@ -58,32 +58,32 @@ function configure_collectd_plugin() {
   shift
   local collectd_plugin_conf="$collectd_conf_plugins_dir/${collectd_plugin}.conf"
   # Create a temporary file for the collectd plugin configration
-  local collectd_plugin_conf_tmp=`mktemp "${collectd_plugin_conf}.XXXXXXXXXX"`
+  local collectd_plugin_conf_tmp=`sudo mktemp "${collectd_plugin_conf}.XXXXXXXXXX"`
   add_mktemp_file $collectd_plugin_conf_tmp
 
-  cat >$collectd_plugin_conf_tmp <<EOF
+  sudo dd of="$collectd_plugin_conf_tmp" <<EOF
 # Generated by BASE collectd RightScript
 LoadPlugin "$collectd_plugin"
 EOF
 
   # Add each configuration option to the collectd plugin configuration if there are any
   if [[ $# -ne 0 ]]; then
-    cat >>$collectd_plugin_conf_tmp <<EOF
+    sudo dd oflag=append conv=notrunc of="$collectd_plugin_conf_tmp" <<EOF
 
 <Plugin "$collectd_plugin">
 EOF
 
     for option in "$@"; do
-      echo "  $option" >>$collectd_plugin_conf_tmp
+      sudo bash -c "echo '  $option' >> $collectd_plugin_conf_tmp"
     done
-    echo '</Plugin>' >>$collectd_plugin_conf_tmp
+    sudo bash -c "echo '</Plugin>' >> $collectd_plugin_conf_tmp"
   fi
 
   # Overwrite and backup the collectd plugin configration if it has changed
   if run_check_write_needed $collectd_plugin_conf $collectd_plugin_conf_tmp; then
-    chmod 0644 $collectd_plugin_conf_tmp
-    [[ -f $collectd_plugin_conf ]] && cp --archive $collectd_plugin_conf "${collectd_plugin_conf}.`date -u +%Y%m%d%H%M%S`"
-    mv --force $collectd_plugin_conf_tmp $collectd_plugin_conf
+    sudo chmod 0644 $collectd_plugin_conf_tmp
+    [[ -f $collectd_plugin_conf ]] && sudo cp --archive $collectd_plugin_conf "${collectd_plugin_conf}.`date -u +%Y%m%d%H%M%S`"
```

Comment:
Are we sure that `$collectd_plugin_conf` can be readable with the rightlink user?  If not, we should put sudo around the check:
```bash
sudo [ -f $collectd_plugin_conf ] && sudo cp --archive $collectd_plugin_conf "${collectd_plugin_conf}.`date -u +%Y%m%d%H%M%S`"
```